### ISSUE TITLE
Not able to setup cluster over encrypted overlay network in docker swarm mode

### ISSUE THREAD

**Issue by jwillmer** [docker-library/rabbitmq#494]
When I setup a network like: 
```
networks:
  rabbitmq-cluster:
    driver: overlay
    driver_opts:
      encrypted: "true"
      attachable: "true"
```

and use a script like this to join the rabbitmq nodes:
```
#!/bin/bash

set -e

# Start RMQ from entry point.
# This will ensure that environment variables passed
# will be honored
/usr/local/bin/docker-entrypoint.sh rabbitmq-server -detached

# Do the cluster dance
rabbitmqctl stop_app
rabbitmqctl join_cluster rabbit@rabbitmq1
...
```

I get the following error:
```
Clustering node rabbit@rabbitmq2 with rabbit@rabbitmq1
Error: unable to perform an operation on node 'rabbit@rabbitmq1'. Please see diagnostics information and suggestions below.

Most common reasons for this are:

 * Target node is unreachable (e.g. due to hostname resolution, TCP connection or firewall issues)
 * CLI tool fails to authenticate with the server (e.g. due to CLI tool's Erlang cookie not matching that of the server)
 * Target node is not running

In addition to the diagnostics info below:

 * See the CLI, clustering and networking guides on https://rabbitmq.com/documentation.html to learn more
 * Consult server logs on node rabbit@rabbitmq1
 * If target node is configured to use long node names, don't forget to use --longnames with CLI tools

DIAGNOSTICS
===========

attempted to contact: [rabbit@rabbitmq1]

rabbit@rabbitmq1:
  * unable to connect to epmd (port 4369) on rabbitmq1: timeout (timed out)


Current node details:
 * node name: 'rabbitmqcli-989-rabbit@rabbitmq2'
 * effective user's home directory: /var/lib/rabbitmq
 * Erlang cookie hash: Kw2YuOlqWicb32+I7Ofxyw==
```

I know about the docker swarm ports:

- TCP port 2377 for cluster management communications
- TCP and UDP port 7946 for communication among nodes
- UDP port 4789 for overlay network traffic
- ENCRYPTED ip protocol 50 (ESP)

And this is my firewall/iptable setup on all servers belonging to the cluster:

```
ufw default deny incoming
ufw default allow outgoing

ufw allow proto tcp to 0.0.0.0/0 port 22
ufw allow http
ufw allow https

ufw allow from 192.168.8.1/24 to any proto tcp port 2377
ufw allow from 192.168.8.1/24 to any proto tcp port 7946
ufw allow from 192.168.8.1/24 to any proto udp port 7946
ufw allow from 192.168.8.1/24 to any proto udp port 4789

ufw enable

iptables -A INPUT -p 50 -j ACCEPT
```

I'm using image: `rabbitmq:3.8-management`. **When I do not enable the encryption on the network the rabbit cluster can successfully join.** 

Anyone had the same problem, knows why this is a problem or how to fix it?

**Comment by tianon** [comment]
This sounds like it's probably an issue with your underlying environment more than with RabbitMQ -- I'd suggest trying to debug encrypted overlay networks on your setup with something simpler like just running two containers and trying to ping between them (or use `nc` to set up a simple TCP server and route a little traffic).

For help doing that, I'd also recommend going to a dedicated support forum, such as [the Docker Community Forums](https://forums.docker.com/), [the Docker Community Slack](https://dockr.ly/slack), or [Stack Overflow](https://stackoverflow.com/search?tab=newest&q=docker).

**Comment by jwillmer** [comment]
The problem is that I already setup a mongodb cluster successfully. Only rabbitmq has problems connecting in an encrypted network. I agree that this might not be directly a problem of rabbitmq but I hoped that it is a known issue and there are available fixes for it.

**Comment by jwillmer** [comment]
Solved it by adding `ufw allow from 192.168.8.1/24 to any proto esp`
